src/settings/convertSetting.ts

```diff
@@ -0,0 +1,19 @@
+import { ConversionError } from "../errors/conversionError";
+import { EditorSettingConverter } from "./converter";
+import { EditorSetting } from "./types";
+
+export const convertSetting = (
```

Comment:
Yeah, coming back to this, I think this should really be called `convertEditorSetting`. It's a bit confusing at first read!

Reply:
Refactored and specified it as follows:
- **editorConfiguration** is the entire configuration file, e.g. `.vscode/settings.json`. Therefore `findEditorConfiguration` and `writeEditorConfigConversionResults` handle the entire config
- **editorSettings** are entries of the editor configuration prefixed with `editor.`, thus `convertEditorSettings` and `reportEditorSettingConversionResults` only handle this subset

Reply:
Also renamed the property `settingName` in `EditorSetting` to `editorSettingName`. However, I am a bit unsure whether `settingName` would suffice.
@JoshuaKGoldberg What do you think?

Reply:
Heh, and we could probably switch those names and still see it as all making sense... For now, I think it looks nice as is - erring on the side of caution given how many areas have confusingly similar names. Unless you start to feel strongly about going to `settingName`, let's leave it as `editorSettingName`? 🤷‍♂